integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -223,11 +223,14 @@ protected OpenLineage.JobBuilder buildJob(QueryExecution queryExecution) {
     if (node instanceof WholeStageCodegenExec) {
       node = ((WholeStageCodegenExec) node).child();
     }
+
+
+    String name = eventEmitter.getOverwriteName().isPresent() ? eventEmitter.getOverwriteName().get() : sparkContext.appName();
     return openLineage
         .newJobBuilder()
         .namespace(this.eventEmitter.getJobNamespace())
         .name(
-            sparkContext.appName().replaceAll(CAMEL_TO_SNAKE_CASE, "_$1").toLowerCase(Locale.ROOT)
```

Comment:
Consider extracting this to a method with some doc provided to describe what's happening here.

Reply:
done